src/ScriptCs/Program.cs

```diff
@@ -12,41 +10,51 @@ internal static class Program
         [LoaderOptimizationAttribute(LoaderOptimization.MultiDomain)]
         private static int Main(string[] args)
         {
-            SetProfile();
-            
-            ArgumentParseResult arguments;
-            var console = new ScriptConsole();
+            ProfileOptimizationShim.SetProfileRoot(Path.GetDirectoryName(typeof(Program).Assembly.Location));
+            ProfileOptimizationShim.StartProfile(typeof(Program).Assembly.GetName().Name + ".profile");
+
+            var nonScriptArgs = args.TakeWhile(arg => arg != "--").ToArray();
+            var scriptArgs = args.Skip(nonScriptArgs.Length + 1).ToArray();
+
+            ScriptCsArgs commandArgs;
             try
             {
-                var parser = new ArgumentHandler(new ArgumentParser(), new ConfigFileParser(console), new FileSystem());
-                arguments = parser.Parse(args);
+                commandArgs = ScriptCsArgs.Parse(nonScriptArgs);
             }
-            catch(Exception ex)
+            catch (Exception ex)
             {
-                console.WriteLine(ex.Message);
-                var options = new ArgUsageOptions { ShowPosition = false, ShowType = false };
-                var usage = ArgUsage.GetUsage<ScriptCsArgs>(options: options);
-                console.WriteLine(usage);
+                Console.WriteLine(ex.Message);
+                Console.WriteLine(ScriptCsArgs.GetUsage());
                 return 1;
             }
 
-            var scriptServicesBuilder = ScriptServicesBuilderFactory.Create(arguments.CommandArguments, arguments.ScriptArguments);
-            var factory = new CommandFactory(scriptServicesBuilder);
-            var command = factory.CreateCommand(arguments.CommandArguments, arguments.ScriptArguments);
-            return (int)command.Execute();
-        }
+            if (commandArgs.Help)
+            {
+                Console.WriteLine(ScriptCsArgs.GetUsage());
```

Comment:
I've really no idea. It's a nice side effect of the PR though :stuck_out_tongue_winking_eye: